### PR TITLE
CXP-1295: Endpoint PUT product mapping schema

### PR DESCRIPTION
**Pull request by masacc**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/ServiceAPI/Command/UpdateCatalogProductMappingSchemaCommand.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+use Akeneo\Catalogs\Infrastructure\Validation as CatalogAssert;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @codeCoverageIgnore
+ */
+final class UpdateCatalogProductMappingSchemaCommand implements CommandInterface
```
Comment (willy-ahva) [inline components/catalogs/back/src/ServiceAPI/Command/UpdateCatalogProductMappingSchemaCommand.php:16]:
```suggestion
final class UpdateProductMappingSchemaCommand implements CommandInterface
```

File: components/catalogs/back/src/ServiceAPI/Query/GetCatalogProductMappingSchemaQuery.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @implements QueryInterface<object>
+ * @codeCoverageIgnore
+ */
+final class GetCatalogProductMappingSchemaQuery implements QueryInterface
```
Comment (willy-ahva) [inline components/catalogs/back/src/ServiceAPI/Query/GetCatalogProductMappingSchemaQuery.php:16]:
```suggestion
final class GetProductMappingSchemaQuery implements QueryInterface
```

**Review by willy-ahva — COMMENTED** [review]
GTM but let me a moment to check about 200 / 204 return status. Good job!

**Comment by willy-ahva** [comment]
> GTM but let me a moment to check about 200 / 204 return status. Good job!

Seen with Mégane, let's do a 204. Ty

**Review by guillaumefer-akeneo — APPROVED** [review]

**Review by eoliveauakeneo — APPROVED** [review]

**Pull request merged**